### PR TITLE
Add ADN Website

### PR DESCRIPTION
**Pull request by Skytowz**

### CODE REVIEW

File: src/pages/ADN/main.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import { pageInterface } from '../pageInterface';
+
+const style = require('!to-string-loader!css-loader!less-loader!./style.less');
+
+export const ADN: pageInterface = {
+  name: 'ADN',
+  domain: 'https://animationdigitalnetwork.fr/',
```
Comment (henrik9999) [inline src/pages/ADN/main.ts:7]: without ending /

**Review by henrik9999 — CHANGES_REQUESTED** [review]

File: src/pages/ADN/main.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      return j.$('div[data-testid="player-content"] h1 a').text().trim();
+    },
+    getIdentifier(url) {
+      return utils.urlPart(url, 4) || '';
+    },
+    getOverviewUrl(url) {
+      return `${ADN.domain}/video/${ADN.sync.getIdentifier(url)}`;
+    },
+    getEpisode(url) {
+      return Number(utils.urlPart(url, 5) ? utils.urlPart(url, 5).split('-')[2] : '');
```
Comment (henrik9999) [inline src/pages/ADN/main.ts:29]: probably better to use an regex like episode-(\d+)

File: src/pages/ADN/main.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    list: {
+      offsetHandler: false,
+      elementsSelector() {
+        return j.$('div[data-testid="default-layout"] ul li[itemtype] > div');
+      },
+      elementUrl(selector) {
+        return utils.absoluteLink(selector.find('a').attr('href'), ADN.domain);
+      },
+      elementEp(selector) {
+        return Number(selector.find('h3').first().text().split(' ').pop() || '');
```
Comment (henrik9999) [inline src/pages/ADN/main.ts:60]: would be better to use the getEpisode out of sync

**Review by henrik9999 — CHANGES_REQUESTED** [review]

File: src/pages/ADN/main.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
+import { pageInterface } from '../pageInterface';
+
+const style = require('!to-string-loader!css-loader!less-loader!./style.less');
```
Comment (lolamtisch) [inline src/pages/ADN/main.ts:3]: The build has problems with this. Add it like on the other pages.

https://github.com/MALSync/MALSync/blob/master/src/pages/Gogoanime/main.ts#L85

**Review by lolamtisch — CHANGES_REQUESTED** [review]

File: src/pages/ADN/main.ts (hunk 4 above)
Comment (Skytowz) [reply src/pages/ADN/main.ts:3]: Yes, I saw that, I was trying to remove some warning :/

File: src/pages/ADN/main.ts (hunk 2 above)
Comment (Skytowz) [reply src/pages/ADN/main.ts:29]:
```suggestion
      return Number(utils.urlPart(url, 5).split('-')[2]);
```
Does this work for you?

File: src/pages/ADN/main.ts (hunk 3 above)
Comment (Skytowz) [reply src/pages/ADN/main.ts:60]:
```suggestion
        return ADN.sync.getEpisode(selector.find('a').attr('href') || '');
```
Like this?

Comment (henrik9999) [reply src/pages/ADN/main.ts:60]: Yes

File: src/pages/ADN/main.ts (hunk 2 above)
Comment (henrik9999) [reply src/pages/ADN/main.ts:29]: No using regex is better

File: src/pages/ADN/main.ts [hunk 5]
```diff
@@ -26,7 +24,10 @@ export const ADN: pageInterface = {
       return `${ADN.domain}/video/${ADN.sync.getIdentifier(url)}`;
     },
     getEpisode(url) {
-      return Number(utils.urlPart(url, 5) ? utils.urlPart(url, 5).split('-')[2] : '');
+      const temp = url.match(/episode-(\d+)/i);
+      con.log(temp);
```
Comment (henrik9999) [inline src/pages/ADN/main.ts:28]: Remove the log

**Review by henrik9999 — CHANGES_REQUESTED** [review]

File: src/pages/ADN/main.ts (hunk 5 above)
Comment (Skytowz) [reply src/pages/ADN/main.ts:28]:
```suggestion
```

File: src/pages/ADN/main.ts [hunk 6]
```diff
@@ -26,7 +24,10 @@ export const ADN: pageInterface = {
...
+      if (!temp) return NaN;
```
[the remark this answers is not in the document]
Comment (Skytowz) [reply src/pages/ADN/main.ts:29]:
```suggestion
      if (!temp) return 1;
```

File: src/pages/ADN/tests.json [hunk 7]
```diff
@@ -0,0 +1,52 @@
+{
+  "title": "ADN",
+  "url": "https://animationdigitalnetwork.fr",
+  "enabled": false,
```
Comment (henrik9999) [inline src/pages/ADN/tests.json:4]: Test is disabled

**Review by henrik9999 — CHANGES_REQUESTED** [review]

File: src/pages/ADN/tests.json (hunk 7 above)
Comment (Skytowz) [reply src/pages/ADN/tests.json:4]:
```suggestion
  "enabled": true,
```

**Comment by Skytowz** [comment]
Sorry, the test were done by my friend...

File: src/pages/ADN/main.ts [hunk 8]
```diff
@@ -0,0 +1,88 @@
+import { pageInterface } from '../pageInterface';
+
+export const ADN: pageInterface = {
+  name: 'ADN',
+  domain: 'https://animationdigitalnetwork.fr',
+  languages: ['French'],
+  type: 'anime',
+  isSyncPage(url) {
+    if ($('div[data-testid="video-player"]').length) return true;
```
Comment (lolamtisch) [inline src/pages/ADN/main.ts:9]: By using the Boolean operator here and every where else you dont need to use a if.
`return Boolean($('div[data-testid="video-player"]').length);`

**Review by lolamtisch — CHANGES_REQUESTED** [review]

File: src/pages/ADN/style.less [hunk 9]
```diff
@@ -0,0 +1,39 @@
+@import './../pages';
```
[the remark this answers is not in the document]
Comment (Skytowz) [reply src/pages/ADN/style.less:1]: What ep do you want to be highlight? the actual ep or the ep indicated in our list?

[the remark this answers is not in the document]
Comment (lolamtisch) [reply src/pages/ADN/style.less:1]: You only need to style the class. It is handled by an option if all eps should be highlighted

image

[the remark this answers is not in the document]
Comment (Skytowz) [reply src/pages/ADN/style.less:1]: Okay, I think I understand what you mean by remove the default.
But if I do like you told me I got a sort of ambiguous situation between the style of the website and mine
image
Like by default ADN put a blue border on the actual

[the remark this answers is not in the document]
Comment (Skytowz) [reply src/pages/ADN/style.less:1]: image
Like this would not be better?

File: src/pages/ADN/meta.json [hunk 10]
```diff
@@ -0,0 +1,5 @@
+{
+  "urls": {
+    "match": ["*://animationdigitalnetwork.fr/*"]
```
Comment (lolamtisch) [inline src/pages/ADN/meta.json:3]: The search shortcut url is missing

example: https://github.com/MALSync/MALSync/blob/master/src/pages/An1me/meta.json#L2

File: src/pages/ADN/style.less [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
```
Comment (lolamtisch) [inline src/pages/ADN/style.less:1]: The styling for 'add to mal' is still broken. But everything else looks good

image

**Review by lolamtisch — CHANGES_REQUESTED** [review]

**Review by lolamtisch — APPROVED** [review]

**Pull request merged**